File: testdata/complexity_sample.go
package main

// Complexity 1: no decision points
func simpleFunc() int {
	return 42
}

// Complexity 4: if + else if + for + &&
func complexFunc(x int, items []string) int {
	if x > 0 && x < 100 {
		return 1
	} else if x >= 100 {
		return 2
	}
	for _, item := range items {
		_ = item
	}
	return 0
}

// Complexity 5: switch with 4 cases
func switchFunc(op string) int {
	switch op {
	case "add":
		return 1
	case "sub":
		return 2
	case "mul":
		return 3
	default:
		return 0
	}
}

// Complexity 3: nested if + ||
func nestedFunc(a, b bool) string {
	if a {
		if b || a {
			return "both"
		}
		return "a"
	}
	return "none"
}

File: testdata/complexity_sample.py
# testdata/complexity_sample.py

# Complexity 1
def simple_func():
    return 42

# Complexity 6: if + and + elif + for + if + base (or not counted)
def complex_func(x, items):
    if x > 0 and x < 100:
        return 1
    elif x >= 100:
        return 2
    for item in items:
        if item or x:
            pass
    return 0

# Complexity 4: while + except + for + base
def loop_func(n):
    while n > 0:
        n -= 1
    try:
        result = 1 / n
    except ZeroDivisionError:
        result = 0
    for i in range(n):
        pass
    return result

# Complexity 3: if + comprehension-if + base
def comprehension_func(items):
    if items:
        return [x for x in items if x > 0]
    return []
